File: 2021/6/day6.py
with open("day6_input.txt") as input:
    l = [int(d) for d in input.read().split(",")]

for i in range(30):
    print(str(len(l)) + ",")
    for j in range(len(l)):
        l[j] -= 1
        if l[j] == -1:
            l[j] = 6
            l.append(8)

print(len(l))

File: 2023/6/day6.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func getMargin(time, distance int) (margin int) {
	for i := 0; i < time; i++ {
		if i*(time-i) > distance {
			margin++
		}
	}
	return
}

func main() {
	input, _ := os.ReadFile("input.txt")

	data := strings.Fields(string(input))
	times := data[1 : len(data)/2]
	distances := data[len(data)/2+1:]

	product := 1
	for race := range times {
		time, _ := strconv.Atoi(times[race])
		distance, _ := strconv.Atoi(distances[race])
		product *= getMargin(time, distance)
	}

	time, _ := strconv.Atoi(strings.Join(times, ""))
	distance, _ := strconv.Atoi(strings.Join(distances, ""))

	fmt.Println(product)
	fmt.Println(getMargin(time, distance))
}
